modules/imgproc/src/hal_replacement.hpp

```diff
@@ -856,6 +856,26 @@ inline int hal_ni_threshold(const uchar* src_data, size_t src_step, uchar* dst_d
 #define cv_hal_threshold hal_ni_threshold
 //! @endcond
 
+/**
+   @brief Performs threshold filtering using threshold estimated by Otsu algorithm
+   @param src_data Source image data
+   @param src_step Source image step
+   @param dst_data Destination image data
+   @param dst_step Destination image step
+   @param width Source image width
+   @param height Source image height
+   @param depth Depths of source and destination image
+   @param cn Number of channels
+   @param maxValue Value assigned to the pixels for which the condition is satisfied
+   @param thresholdType Thresholding type
+   @param thresh Calculated threshold value
+*/
+inline int hal_ni_threshold_otsu(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, int depth, int cn, double maxValue, int thresholdType, double* thresh) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
Do we really need this function? If we really need Otsu in HAL, I would simply add 

```
int hal_ni_get_otsu_threshold(const uchar* src_data, size_t src_step, int width, int height, int depth, double* thresh) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

that is, I would add the function that computes value of Otsu threshold, and then simply call `hal_ni_threshold()` with the computed threshold value.